sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -71,7 +71,8 @@ case class CreateArray(children: Seq[Expression]) extends Expression {
       s"""
         final ArrayData ${ev.value} = new $arrayClass($values);
         this.$values = null;
-      """)
+      """,
+      isNull = "false")
```

Comment:
with this, I think we can remove the `final boolean ${ev.isNull} = false;` in the generated code.